### PR TITLE
Sanjo/new vault2

### PR DESCRIPTION
**Pull request by SanjoDeundiak**

### CODE REVIEW

File: implementations/rust/ockam/ockam_vault_core/src/open_close_vault.rs [hunk 1]
```diff
@@ -2,6 +2,12 @@ use crate::vault::Vault;
 use ockam_error::OckamResult;
 
 pub trait OpenCloseVault: Sized {
+    type InnerVault;
+
+    fn get_data_mut(&mut self) -> &mut Self::InnerVault;
```
Comment (mikelodder7) [inline implementations/rust/ockam/ockam_vault_core/src/open_close_vault.rs:7]: I'm not understanding the purpose for these functions.

File: implementations/rust/ockam/ockam_vault/src/lib.rs [hunk 2]
```diff
@@ -1,71 +1,10 @@
...
-        let vault = vault.open().unwrap();
-
-        let hash = vault.sha256(&[0u8; 32]).unwrap();
-        assert_eq!(
-            hex::encode(hash),
-            "66687aadf862bd776c8fc18b8e9f8e20089714856ee233b3902a591d0d5f2925"
-        );
-    }
-}
+pub extern crate ockam_vault_core;
```
Comment (mikelodder7) [inline implementations/rust/ockam/ockam_vault/src/lib.rs:1]: Do we want to alias this crate or perhaps only export a subset of the crate?

Comment (SanjoDeundiak) [reply implementations/rust/ockam/ockam_vault/src/lib.rs:1]: @mrinalwadhwa I think it's question to you

File: implementations/rust/ockam/ockam_vault_core/src/open_close_vault.rs (hunk 1 above)
Comment (SanjoDeundiak) [reply implementations/rust/ockam/ockam_vault_core/src/open_close_vault.rs:7]: @mikelodder7 it allows to use vaults like this:
```rs
let mut vault = SoftwareVault::default();
let mut vault = vault.open().unwrap();
let mut attributes = SecretAttributes {
    stype: SecretType::Curve25519,
    persistence: SecretPersistence::Ephemeral,
    length: CURVE25519_SECRET_LENGTH,
};

let res = vault.secret_generate(attributes);
```

So, SoftwareVault implements OpenCloseVault trait, but none of other vault traits. InnerVault is SoftwareVaultImpl and implements all the other vault traits (like sign, verify, encrypt, etc.)

**Review by mikelodder7 — APPROVED** [review]

**Review by mrinalwadhwa — APPROVED** [review]

**Pull request merged**